pkg/syncer/syncer.go

```diff
@@ -299,6 +324,55 @@ func (c *Controller) handleErr(err error, i interface{}) {
 	klog.Errorf("Dropping key %q after failed retries: %v", i, err)
 }
 
+const nsDelimiter = "--"
+const nsTemplate = "kcp" + nsDelimiter + "%s" + nsDelimiter + "%s"
+
+type nameTransformer struct {
+	clusterName string
+	namespace   string
+}
+
+func (n *nameTransformer) toCluster() string {
+	return fmt.Sprintf(nsTemplate, n.clusterName, n.namespace)
+}
+
+func toKcp(namespace string) (*nameTransformer, error) {
+	segments := strings.Split(namespace, nsDelimiter)
+
+	if len(segments) != 3 {
+		return nil, fmt.Errorf("Invalid name: %s", namespace)
+	}
+
+	return &nameTransformer{
+		clusterName: segments[1],
+		namespace:   segments[2],
+	}, nil
+}
+
+func transformForCluster(gvr schema.GroupVersionResource, meta metav1.Object) (string, string, error) {
```

Comment:
I'm a bit worried about making that a load-bearing assumption, in case we add another cluster-scoped resource in the future (not sure what, but you never know), and this subtly stops being true.

Reply:
On reflection, I realize that my original comment focused too much on how namespaces should be treated specially. An empty namespace signifies a cluster-scoped resource, and keying on that instead of gvr would provide the desired name transformation for all cluster-scoped resources (not just namespaces).